File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/fetchaggregationandgroupby10.vb
' <snippetfetchaggregationandgroupby10>


 ' *****************************************************************************************************************
 '                FetchXML      groupby2   Aggregate 10
 ' *****************************************************************************************************************
 ' Fetch the number of opportunities each manager's direct reports 
 ' own using a groupby within a link-entity.
 Dim groupby2 As String = " " &amp; ControlChars.CrLf &amp; _
     "                    <fetch distinct='false' mapping='logical' aggregate='true'> " &amp; ControlChars.CrLf &amp; _
     "                        <entity name='opportunity'> " &amp; ControlChars.CrLf &amp; _
     "                           <attribute name='name' alias='opportunity_count' aggregate='countcolumn' /> " &amp; ControlChars.CrLf &amp; _
     "                           <link-entity name='systemuser' from='systemuserid' to='ownerid'>" &amp; ControlChars.CrLf &amp; _
     "                               <attribute name='parentsystemuserid' alias='managerid' groupby='true' />" &amp; ControlChars.CrLf &amp; _
     "                           </link-entity> " &amp; ControlChars.CrLf &amp; _
     "                        </entity> " &amp; ControlChars.CrLf &amp; _
     "                    </fetch>"

 Dim groupby2_result As EntityCollection = _serviceProxy.RetrieveMultiple(New FetchExpression(groupby2))

 For Each c In groupby2_result.Entities

       Dim aggregate10a? As Integer = CType((CType(c("opportunity_count"), AliasedValue)).Value, Integer?)
       Console.WriteLine("Count of all opportunities: " &amp; aggregate10a &amp; vbLf)
 Next c
 Console.WriteLine("===============================")

' </snippetfetchaggregationandgroupby10>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/fetchaggregationandgroupby10.cs
// <snippetfetchaggregationandgroupby10>


 // *****************************************************************************************************************
 //                FetchXML      groupby2   Aggregate 10
 // *****************************************************************************************************************
 // Fetch the number of opportunities each manager's direct reports 
 // own using a groupby within a link-entity.
 string groupby2 = @" 
 <fetch distinct='false' mapping='logical' aggregate='true'> 
     <entity name='opportunity'> 
        <attribute name='name' alias='opportunity_count' aggregate='countcolumn' /> 
        <link-entity name='systemuser' from='systemuserid' to='ownerid'>
            <attribute name='parentsystemuserid' alias='managerid' groupby='true' />
        </link-entity> 
     </entity> 
 </fetch>";

 EntityCollection groupby2_result = _serviceProxy.RetrieveMultiple(new FetchExpression(groupby2));

 foreach (var c in groupby2_result.Entities)
 {

       int? aggregate10a = (int?)((AliasedValue)c["opportunity_count"]).Value;
       System.Console.WriteLine("Count of all opportunities: " + aggregate10a + "\n");
 }
 System.Console.WriteLine("===============================");

// </snippetfetchaggregationandgroupby10>